### PR TITLE
test(widgetbook): :rotating_light: testing theme addons  🚨

### PR DESCRIPTION
**Pull request by mulieriq**
- Added test for custom theme

### List of issues which are fixed by the PR
*You must list at least one issue.*

### Screenshots
*If applicable, add screenshots to help explain the changes.*

### Checklist

- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on [Discord].

[CLA]: https://docs.google.com/forms/d/e/1FAIpQLScuRfjUzENsLsmQgqZlGLxMKbFi7zuXoPARyXytoyQrq7ntUw/viewform
[Discord]: https://discord.com/invite/zT4AMStAJA

### CODE REVIEW

File: packages/widgetbook/test/src/addons/custom_theme_addon_test.dart [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+import 'package:widgetbook/src/workbench/renderer.dart';
+
+import 'utils/custom_app_theme.dart';
+import 'utils/widget_wrapper.dart';
+
+void main() {
+  late Renderer sut;
+  final rendererKey = GlobalKey();
+
+  setUpAll(() {
```
Comment (jenshor) [inline packages/widgetbook/test/src/addons/custom_theme_addon_test.dart:14]: I think we should rarely use `setUpAll` and use `setUp` instead since obtaining a fresh instance will guarantee that there are no side effects from other tests

File: packages/widgetbook/test/src/addons/custom_theme_addon_test.dart [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: const [
+              Text(
+                'This is the home page',
+              ),
+            ],
+          ),
+        ),
+      ),
```
Comment (jenshor) [inline packages/widgetbook/test/src/addons/custom_theme_addon_test.dart:40]: Is it necessary to have such a complex Widget here? I'd just use a `ColoredBox` or colored `Container` here instead.

File: packages/widgetbook/test/src/addons/custom_theme_addon_test.dart [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    );
+  });
+
+  group('$Renderer', () {
+    testWidgets(
+      'renders correctly',
```
Comment (jenshor) [inline packages/widgetbook/test/src/addons/custom_theme_addon_test.dart:46]: It's okay to have such a test, but this is unrelated to the `CustomThemeAddon`

File: packages/widgetbook/test/src/addons/custom_theme_addon_test.dart [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+                'This is the home page',
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  });
+
+  group('$Renderer', () {
```
Comment (jenshor) [inline packages/widgetbook/test/src/addons/custom_theme_addon_test.dart:44]: In general, this group should be called `CustomThemeAddon` (or this test file should be renamed)

File: packages/widgetbook/test/src/addons/custom_theme_addon_test.dart [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+          addOnProviderWrapper(
+            child: sut,
+          ),
+        );
+
+        expect(find.byKey(rendererKey), findsOneWidget);
+      },
+    );
+    testWidgets(
+      'renders one Text Widget',
```
Comment (jenshor) [inline packages/widgetbook/test/src/addons/custom_theme_addon_test.dart:58]: Also unrelated to the addon. This is pretty much the same test as the one above.

File: packages/widgetbook/test/src/addons/custom_theme_addon_test.dart [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+            child: sut,
+          ),
+        );
+
+        expect(find.byType(Text), findsOneWidget);
+      },
+    );
+
+    testWidgets(
+      'renders the correct text',
```
Comment (jenshor) [inline packages/widgetbook/test/src/addons/custom_theme_addon_test.dart:71]: This is the same test as the one above.

File: packages/widgetbook/test/src/addons/custom_theme_addon_test.dart [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+        );
+      },
+    );
+
+    testWidgets(
+      'can de-activate $customTheme2 when $ThemeSettingProvider..tapped(customTheme2) is called',
+      (WidgetTester tester) async {
+        final provider = ThemeSettingProvider(
+          CustomThemeSetting<AppThemeData>(
+            activeThemes: {customTheme},
```
Comment (jenshor) [inline packages/widgetbook/test/src/addons/custom_theme_addon_test.dart:147]: you should have customTheme2 as an active theme here because otherwise you'll also test activating. However, this test is called "de-activate". In addition, you'd also cover the test behavior of the "activate" test with this one. Why do we need the other test at all if this covers the functionality as well? The reason is that we want to test activate and deactivate separately.

File: packages/widgetbook/test/src/addons/utils/widget_wrapper.dart [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+);
+
+final activeFrameBuilder = WidgetbookFrameBuilder(
+  devices: devices,
+);
+
+Widget addOnProviderWrapper({
+  required Widget child,
+  ThemeSettingProvider<AppThemeData>? provider,
+}) =>
```
Comment (jenshor) [inline packages/widgetbook/test/src/addons/utils/widget_wrapper.dart:47]: This will scale bad for adding other AddOn tests but it's fine for now.

**Review by jenshor — CHANGES_REQUESTED** [review]
Already looks good. Just some minor things 👍🏻

File: packages/widgetbook/test/src/addons/utils/widget_wrapper.dart (hunk 8 above)
Comment (mulieriq) [reply packages/widgetbook/test/src/addons/utils/widget_wrapper.dart:47]: Yes I realized this and modified it fit other Addons

File: packages/widgetbook/test/src/addons/custom_theme_addon_test.dart (hunk 4 above)
Comment (mulieriq) [reply packages/widgetbook/test/src/addons/custom_theme_addon_test.dart:44]: Removed

**Comment by mulieriq** [comment]
Closing this pr ->. Already handled in a different pr
